tee-worker/litentry/core/stf-task/receiver/src/handler/assertion.rs

```diff
@@ -175,7 +177,11 @@ where
 			credential.add_proof(&sig, credential.issuance_block_number, &enclave_account);
 
 			if credential.validate().is_err() {
-				error!("failed to validate credential");
+				let error_msg = "failed to validate credential";
+				error!("	[BuildAssertion] : {}", error_msg);
+
+				self.on_failure(to_vcmp_error(error_msg));
```

Comment:
calling `on_failure` within `on_success` sounds strange, can we have better handling? (ignore this if we can't)

Reply:
At first, I wanted to write another method to deal with such errors, but later I felt a little redundant because I felt that `on_failure` needs to handle all errors from `on_process` and `on_success`, so this method is not written, but directly call `on_failure` method in `on_success`. Please correct me if I understand this wrong.

Reply:
Thanks - if we don't have better ways to handle it, let's just leave it as it is right now :)